lib/main.dart

```diff
@@ -110,37 +112,15 @@ void _setupOfflineListenLogHelper() {
 }
 
 Future<void> _setupDownloadsHelper() async {
-  GetIt.instance.registerSingleton(DownloadsHelper());
-  final downloadsHelper = GetIt.instance<DownloadsHelper>();
-
-  // We awkwardly cache this value since going from 0.6.14 -> 0.6.16 will switch
-  // hasCompletedBlurhashImageMigration despite doing a fixed migration
-  final shouldRunBlurhashImageMigrationIdFix =
-      FinampSettingsHelper.finampSettings.shouldRunBlurhashImageMigrationIdFix;
-
-  if (!FinampSettingsHelper.finampSettings.hasCompletedBlurhashImageMigration) {
-    await downloadsHelper.migrateBlurhashImages();
-    FinampSettingsHelper.setHasCompletedBlurhashImageMigration(true);
-  }
+  GetIt.instance.registerSingleton(IsarDownloads());
+  final isarDownloads = GetIt.instance<IsarDownloads>();
 
-  if (shouldRunBlurhashImageMigrationIdFix) {
-    await downloadsHelper.fixBlurhashMigrationIds();
-    FinampSettingsHelper.setHasCompletedBlurhashImageMigrationIdFix(true);
+  if (!FinampSettingsHelper.finampSettings.hasCompletedIsarDownloadsMigration) {
+    await isarDownloads.migrateFromHive();
+    FinampSettingsHelper.setHasCompletedIsarDownloadsMigration(true);
   }
-}
-
-Future<void> _setupDownloader() async {
-  GetIt.instance.registerSingleton(DownloadUpdateStream());
-  GetIt.instance<DownloadUpdateStream>().setupSendPort();
-
-  WidgetsFlutterBinding.ensureInitialized();
-  await FlutterDownloader.initialize(debug: true);
-
-  // flutter_downloader sometimes crashes when adding downloads. For some
-  // reason, adding this callback fixes it.
-  // https://github.com/fluttercommunity/flutter_downloader/issues/445
-
-  FlutterDownloader.registerCallback(_DummyCallback.callback);
+  await FileDownloader().configure(globalConfig:(Config.checkAvailableSpace, 500));
+  await FileDownloader().resumeFromBackground();
```

Comment:
This call may trigger status and progress updates, so to ensure you are acting on those correctly, you should move this call to a point _after_ you have established your listener.  There is some logic in the downloader to retry if no lister is registered, but it is probably best to call resumeFromBackground after you have registered the listener.

Reply:
Attaching the listener actually happens just before this in the IsarDownloads() constructer.  It's kind of messy with some elements being in main and some in IsarDownloads, but I believe I have all the FileDownloader() setup calls in the correct order.